packages/jellyfish-api-core/__tests__/category/loan/createLoanScheme.test.ts

```diff
@@ -0,0 +1,106 @@
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloanschemes')
+    const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+    expect(result.length).toStrictEqual(1)
+    expect(result[0]).toStrictEqual(
+      { id: 'scheme1', mincolratio: 200, interestrate: 2.5, default: false }
+    )
+  })
+
+  it('should not createLoanScheme if minColRatio is less than 100', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(99, new BigNumber(2.5), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nminimum collateral ratio cannot be less than 100\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if interestRate is less than 0.01', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(0.00999), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\ninterest rate cannot be less than 0.01\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same minColRatio and interestRate were created before', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(100, new BigNumber(1.5), { id: 'scheme2' }) // NOTE(jingyi2811): Failed because its minColRatio and interestRate are same as default
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme default with same interestrate and mincolratio already exists\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same id was created before', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'default' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme already exist with id default\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is an empty string', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is more than 8 chars long', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '123456789' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should createLoanScheme with utxos', async () => {
```

Comment:
a complete utxo test should assert the rawtx.vin whether contains the input utxo